compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -245,6 +245,28 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
         None
     }
+    fn suggest_missing_writer(
+        &self,
+        rcvr_ty: Ty<'tcx>,
+        args: Option<(&'tcx hir::Expr<'tcx>, &'tcx [hir::Expr<'tcx>])>,
+    ) -> Option<DiagnosticBuilder<'_, ErrorGuaranteed>> {
+        let Some(arg) = args else { return None };
+        let (ty_str, _ty_file) = self.tcx.short_ty_string(rcvr_ty);
+        let mut err =
+            struct_span_err!(self.tcx.sess, arg.0.span, E0599, "cannot write into '{}'", ty_str,);
+        err.span_note(
+            arg.0.span,
+            "must implement 'io::Write', 'fmt::Write', or have a write_fmt method",
+        );
+        if let ExprKind::Lit(_) = arg.0.kind {
+            err.span_help(
+                arg.0.span.shrink_to_lo(),
+                "A writer is needed before this format string",
```

Comment:
```suggestion
                "a writer is needed before this format string",
```

Reply:
this never got applied.

Reply:
ah yes missed that 😞 pushing now... btw I started working on porting this to a diagnostic struct